alphastats/gui/utils/llm_helper.py

```diff
@@ -80,3 +83,75 @@ def llm_connection_test(
 
     except Exception as e:
         return str(e)
+
+
+# Unused now, but could be useful in the future
+def get_display_available_uniprot_info(regulated_features: list) -> dict:
+    """
+    Retrieves and formats UniProt information for a list of regulated features.
+
+    Note: The information is retrieved from the `annotation_store` in the `session_state`, which is filled when the LLM analysis is set up from the anlaysis page.
+
+    Args:
+        regulated_features (list): A list of features for which UniProt information is to be retrieved.
+    Returns:
+        dict: A dictionary where each key is a feature representation and the value is another dictionary
+              containing the 'protein ids' and 'generated text' with formatted UniProt information or an error message, starting with ERROR, so it can be filtered before passing on to the LLM.
+    """
+    text_repr = {}
+    for feature in regulated_features:
+        try:
+            text = format_uniprot_annotation(
+                st.session_state[StateKeys.ANNOTATION_STORE][feature]
+            )
+        except Exception as e:
+            text = f"ERROR: {e}"
+        text_repr[st.session_state[StateKeys.DATASET]._feature_to_repr_map[feature]] = {
+            "protein ids": feature,
+            "generated text": text,
+        }
+    return text_repr
+
+
+# TODO: Write test for this display
+@st.fragment
+def display_uniprot(regulated_genes_dict):
+    c1, c2, c3, c4 = st.columns((1, 1, 3, 1))
+    with c1:
+        if st.button("Select all"):
+            st.session_state[StateKeys.UNIPROT_FIELDS] = ExtractedFields.get_values()
+            st.rerun(scope="fragment")
+    with c2:
```

Comment:
just note: this could also be written as
```
if c2.button("Select none"):
        st.session_state[StateKeys.UNIPROT_FIELDS] = []
        st.rerun(scope="fragment")
```